internal/app/init/main.go

```diff
@@ -175,25 +179,39 @@ func root() (err error) {
 		}
 	}
 
-	go func() {
-		if err := listenForPowerButton(); err != nil {
-			log.Printf("WARNING: power off events will be ignored: %+v", err)
-		}
-	}()
+	poweroffCh, err := listenForPowerButton()
+	if err != nil {
+		log.Printf("WARNING: power off events will be ignored: %+v", err)
+	}
+
+	startupErrCh := make(chan error)
+	termCh := make(chan os.Signal, 1)
+	signal.Notify(termCh, syscall.SIGTERM)
 
 	// Get a handle to the system services API.
 	svcs := system.Services(data)
+	defer svcs.Shutdown()
 
 	// Start containerd.
 	svcs.Start(&services.Containerd{})
 
-	go startSystemServices(data)
-	go startKubernetesServices(data)
+	go startSystemServices(startupErrCh, data)
+	go startKubernetesServices(startupErrCh, data)
+
+	select {
+	case <-poweroffCh:
```

Comment:
Do we need to be able to differentiate between reboot/shutdown?

Reply:
I believe so, but ACPI only provides poweroff here? we can have more channels or channel with some code if we need to have more.